spotify_player/src/cli/client.rs

```diff
@@ -350,3 +362,359 @@ async fn handle_playback_request(
+                                    .as_str(),
+                            );
+                            result.push('\n');
+                        }
+                    } else {
+                        // Remove non following imports as they are now useless
+                        remove_dir_all(dir_path)?;
+                        result.push_str(
+                            format!(
+                                "Not following playlist '{}'. Deleting import...\n",
+                                to_id.id()
+                            )
+                            .as_str(),
+                        )
+                    }
+                }
+
+                Ok(result)
+            }
+        }
+    }
+}
+
+const TRACK_BUFFER_CAP: usize = 100;
+
+/// Imports a playlist into another playlist.
+///
+/// All tracks from the `import_from` playlist are added to the `import_to` playlist if they are not in there already.
+///
+/// The state of `import_from` playlist is stored into a cache file to add/delete the differed tracks between
+/// subsequent imports of the same two playlists.
+async fn playlist_import(
+    client: &Client,
+    import_from: PlaylistId<'static>,
+    import_to: PlaylistId<'static>,
+    delete: bool,
+) -> Result<String> {
+    #[derive(Clone, PartialEq, Eq, Hash, Deserialize, Serialize)]
+    struct TrackData {
+        id: TrackId<'static>,
+        name: String,
+    }
+    // Get playlists' info
+    let (from_tracks, from_name) = match client.playlist_context(import_from.to_owned()).await? {
+        Context::Playlist { tracks, playlist } => (
+            tracks.into_iter().map(|t| TrackData {
+                id: t.id,
+                name: t.name,
+            }),
+            playlist.name,
+        ),
+        _ => unreachable!(),
+    };
+    let (to_tracks, to_name) = match client.playlist_context(import_to.to_owned()).await? {
+        Context::Playlist { tracks, playlist } => (
+            tracks.into_iter().map(|t| TrackData {
+                id: t.id,
+                name: t.name,
+            }),
+            playlist.name,
+        ),
+        _ => unreachable!(),
+    };
+
+    // Get import dir/file
+    let cache_dir = get_cache_folder_path()?;
+    let imports_dir = cache_dir.join("imports");
+    let to_dir = imports_dir.join(import_to.id());
+    let from_file = to_dir.join(import_from.id());
+
+    if !to_dir.exists() {
+        create_dir_all(to_dir)?;
+    }
+    // Construct hash sets of the playlists' track IDs
+    let to_hash_set: HashSet<TrackData> = HashSet::from_iter(to_tracks);
+    let from_hash_set: HashSet<TrackData> = HashSet::from_iter(from_tracks);
+
+    let mut new_tracks_hash_set = &from_hash_set - &to_hash_set;
+
+    let mut result = String::new();
+    result += &format!(
+        "Importing tracks from {}:{} to {}:{}...\n",
+        import_from.id(),
+        from_name,
+        import_to.id(),
+        to_name
+    );
```

Comment:
Should be moved after deleting and before importing new tracks.

Reply:
An example log: 
```
Importing tracks from 7xASNlY8MoDGeEwfA8yxTH:JP to 19W9UXoblc5JYrKCa85mH0:JP...
Deleting previously imported tracks in JP that are not in the current JP...
Deleted tracks:
    1di1C0QI6Y92yZPYn6XYAZ: シルエット
    7dEfa89dZfo6CQPdsgGCF6: Nandemonaiya - movie ver.
Imported tracks:
    5QiiYSr4pJMPOESQznu2IC: WAYO
    6EY5igURAXBCEen56oKUif: Blue
```
which is confusing when outputs "importing ..." then nothing shows up.

Reply:
I had moved the "Importing tracks" to the top because it showed what was importing before it gave more information. Having it at the top gives a better look when you are looking at multiple imports.

If it is not at the top, you see deleted tracks, then where the tracks where deleted from, which to me seems backwards. Is there a different message that would convey that shows import is happening before actual details of the import?

Reply:
I think we can make it like

```
Importing from "playlist A" to "playlist B"...
Deleting previously imported tracks that are no longer available in "playlist A"...
Deleted tracks:
   ....
Importing new tracks since the last import to "playlist B"...
Imported tracks:
   ....
```